src/sagemaker/session.py

```diff
@@ -280,6 +298,19 @@ def _initialize(
             # create a default S3 resource, but only if it needs to fetch from S3
             self.sagemaker_config = load_sagemaker_config(s3_resource=self.s3_resource)
 
+        # after sagemaker_config initialization, update self._default_bucket_name_override if needed
+        self._default_bucket_name_override = resolve_value_from_config(
```

Comment:
nit: can we rename this to `resolve_value_from_defaults_config`

Reply:
sure thing

Reply:
Will tackle later in a separate PR, time permitting